In [2]:
import polars as pl
import numpy as np
import os
from tqdm.notebook import tqdm




# Surreact

In [36]:
column_names = [
    "chest circumference", "waist circumference", "pelvis circumference", "neck circumference",
    "bicep circumference", "thigh circumference", "knee circumference", "arm length", "leg length",
    "calf length", "head circumference", "wrist circumference", "arm span", "shoulder width",
    "torso length", "inner leg length", "height"
]

# Create a DataFrame with the specified column names and empty float values
data = {name: pl.Series(name, [], dtype=pl.Float64) for name in column_names}
measurements = pl.DataFrame(data)

folders = ["test", "train"]

for folder in tqdm(folders):
    files = os.listdir(f"../Surreact-APose/{folder}/bodymeasurements/")
    for file in tqdm(files):
        if not file.endswith(".npy"):
            continue
        current_measurement = map(lambda x: x*100, np.load(f"../Surreact-APose/{folder}/bodymeasurements/"+file, allow_pickle=True))
        measurement_data = {name: [value] for name, value in zip(column_names, current_measurement)}
        current_measurement_dataframe = pl.DataFrame(measurement_data)
        measurements.vstack(current_measurement_dataframe, in_place=True)


mean_series = measurements.mean()

mean_dict = mean_series.to_dict(as_series=False)

for col, mean in mean_dict.items():
    print(f"{col}: {round(mean[0], 2)}")
    



  0%|          | 0/2 [00:00<?, ?it/s]

  0%|          | 0/20000 [00:00<?, ?it/s]

  0%|          | 0/80000 [00:00<?, ?it/s]

chest circumference: 99.71
waist circumference: 88.39
pelvis circumference: 104.81
neck circumference: 36.35
bicep circumference: 29.88
thigh circumference: 52.31
knee circumference: 37.74
arm length: 51.11
leg length: 78.05
calf length: 40.39
head circumference: 60.57
wrist circumference: 17.16
arm span: 175.35
shoulder width: 35.87
torso length: 50.9
inner leg length: 72.87
height: 172.6


chest circumference: 99.71

waist circumference: 88.39

pelvis circumference: 104.81

neck circumference: 36.35

bicep circumference: 29.88

thigh circumference: 52.31

knee circumference: 37.74

arm length: 51.11

leg length: 78.05

calf length: 40.39

head circumference: 60.57

wrist circumference: 17.16

arm span: 175.35

shoulder width: 35.87

torso length: 50.9

inner leg length: 72.87

height: 172.6


# BodyM

In [39]:
column_names = [
    "ankle","arm-length","bicep","calf","chest","forearm","height","hip","leg-length","shoulder-breadth","shoulder-to-crotch","thigh","waist","wrist"

]

# Create a DataFrame with the specified column names and empty float values
data = {name: pl.Series(name, [], dtype=pl.Float64) for name in column_names}
measurements = pl.DataFrame(data)

folders = ["testA","testB","train"]

for folder in tqdm(folders):
    measurements_batch = pl.read_csv(f"../bodym-dataset/{folder}/measurements.csv", has_header=True)
    
    measurements.vstack(measurements_batch.drop("subject_id"), in_place=True)


mean_series = measurements.mean()

mean_dict = mean_series.to_dict(as_series=False)

for col, mean in mean_dict.items():
    print(f"{col}: {round(mean[0], 2)}")


  0%|          | 0/3 [00:00<?, ?it/s]

ankle: 24.1
arm-length: 49.43
bicep: 30.28
calf: 37.23
chest: 101.42
forearm: 26.38
height: 171.61
hip: 102.21
leg-length: 78.1
shoulder-breadth: 35.65
shoulder-to-crotch: 64.65
thigh: 53.83
waist: 89.26
wrist: 16.63


ankle: 24.1

arm-length: 49.43

bicep: 30.28

calf: 37.23

chest: 101.42

forearm: 26.38

height: 171.61

hip: 102.21

leg-length: 78.1

shoulder-breadth: 35.65

shoulder-to-crotch: 64.65

thigh: 53.83

waist: 89.26

wrist: 16.63
